onyo/lib/onyo.py

```diff
@@ -534,165 +512,49 @@ def fsck(self, tests: Optional[list[str]] = None) -> None:
           the validation rulesets defined in ``.onyo/validation/``.
         - "pseudo-keys": verifies that assets do not contain pseudo-key names
         """
-        all_tests = {
-            "clean-tree": self._fsck_clean_tree,
-            "anchors": self._fsck_anchors,
-            "asset-unique": self._fsck_unique_assets,
-            "asset-yaml": self._fsck_yaml,
-            "asset-validity": self._fsck_validation,
-            "pseudo-keys": self._fsck_pseudo_keys,
-        }
-        if tests:
-            # only known tests are accepted
-            if [x for x in tests if x not in all_tests.keys()]:
-                raise ValueError("Invalid test requested. Valid tests are: {}".format(', '.join(all_tests.keys())))
-        else:
-            tests = list(all_tests.keys())
-
-        # run the selected tests
-        for key in tests:
-            # TODO: these should be INFO
-            log.debug(f"'{key}' starting")
-
-            if not all_tests[key]():
-                log.debug(f"'{key}' failed")
-                raise OnyoInvalidRepoError(f"'{self._opdir}' failed fsck test '{key}'")
-
-            log.debug(f"'{key}' succeeded")
+        from .command_utils import fsck as ut_fsck
+        ut_fsck(self, tests)
```

Comment:
This is very nit-picky, and I know that others in the group have different opinions about this, but I want to register that I am personally not a fan of overshortened variable and function names. I feel like the function gets not that often called, that it could not be called `utils_fsck()` :)

Reply:
Thanks a lot :)